core/src/session.cpp

```diff
@@ -21,6 +21,13 @@
 #  include <fstream>
 #endif
 
+namespace irods::http::endpoint_operation
+{
+	// This forward declaration is needed for streaming write operations.
+	// See endpoints/data_objects/src/main.cp for the implementation.
+	auto op_write_streaming(irods::http::session_pointer_type _sess_ptr) -> void;
+} // namespace irods::http::endpoint_operation
```

Comment:
I am curious as to why the implementation ended up in main since it only seems to be used in this file. Not saying it's wrong, just wondering whether it was brought into consideration.

Reply:
The implementation lives in data_objects/src/main.cpp because it's part of the data objects write operation family. The headers for the implementation were already available too. I also wanted to keep session.cpp focused on entry-related logic.

So, it's primarily about keeping related things in the same place.